pennylane_lightning/core/src/simulators/lightning_kokkos/catalyst/LightningKokkosSimulator.cpp

```diff
@@ -404,20 +408,14 @@ void LightningKokkosSimulator::PartialSample(
 
 void LightningKokkosSimulator::Counts(DataView<double, 1> &eigvals,
                                       DataView<int64_t, 1> &counts,
-                                      std::size_t shots) {
+                                      const std::size_t shots) {
```

Comment:
Yea, none of these need to be const. If the outer API requires it, then fine, otherwise it doesn't seem necessary. Fine to leave it be, but in future I'd say no need to make pass-by-value entries const in cases like this.

Reply:
The C++ Core Guidelines [F.16](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const) prefer passing cheaply copied types by value, without `const`. Although IMHO, the more we can mark as `const` the better.

Reply:
> The C++ Core Guidelines [F.16](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const)
> void f3(int x);            // OK: Unbeatable

I think as long as clang-tidy is happy everything's fine here, but I like how they straight up say "unbeatable" :laughing: 
I will go with the true and tried unbeatable recommended by the guideline then.

Reply:
Note that the guide compares const _ref_ to by value, which is an overhead for ints. `const` vs `const int` makes no difference in performance, although personally I don't particular like littering the source code with const annotations unless its with intention.